sympy/combinatorics/permutations.py

```diff
@@ -1243,8 +1243,11 @@ def __mul__(self, other):
         Cycle(1, 3, 2)
 
         """
+        from sympy.combinatorics.perm_groups import PermutationGroup
         a = self.array_form
         # __rmul__ makes sure the other is a Permutation
+        if isinstance(other, PermutationGroup):
+            return [a*i for i in other.generators]
```

Comment:
@asmeurer  I agree. I don't know If I can quote this reference but @smichr  and @asmeurer  please take a look at `cosets` in sage. http://www.sagemath.org/doc/reference/groups/sage/groups/perm_gps/permgroup.html. Even they represent cosets as a list, should we go ahead with this representation for now ?